cpp/monitoring/prometheus/gauge-inl.h

```diff
@@ -12,31 +16,48 @@ PrometheusGauge<LabelTypes...>* PrometheusGauge<LabelTypes...>::New(
   return new PrometheusGauge(name, labels..., help);
 }
 
+
 template <class... LabelTypes>
 PrometheusGauge<LabelTypes...>::PrometheusGauge(
     const std::string& name,
     const typename NameType<LabelTypes>::name&... labels,
     const std::string& help)
     : Gauge<LabelTypes...>(name, labels..., help),
-      values_(&family_, [](::io::prometheus::client::Metric* m) {
-        return m->mutable_gauge();
-      }) {
+      values_(&family_, &::io::prometheus::client::Metric::mutable_gauge,
+              &::io::prometheus::client::Metric::gauge) {
   family_.set_name(name);
   family_.set_help(help);
   family_.set_type(::io::prometheus::client::MetricType::GAUGE);
 }
 
+
+template <class... LabelTypes>
+void PrometheusGauge<LabelTypes...>::Export(std::ostream* os) const {
+  std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
Isn't the ```family_``` also accessed via ```values_```? The only two things that take this mutex are the two export methods, and I have this feeling that for a lock to work, you need both the reader *and the writer* to take it... :wink: